packages/ember-routing/lib/services/router.js

```diff
@@ -6,8 +6,32 @@ import { Service, readOnly } from 'ember-runtime';
 import { shallowEqual, resemblesURL, extractRouteArgs } from '../utils';
 
 /**
-   The Router service is the public API that provides component/view layer
-   access to the router.
+   The Router service provides access to the router.
+   
+   It can be used to check the current route, initiate route transitions, and more.
+   
+   For example, it can be used to transition to a dedicated route if the 
+   application goes offline. Note that this is a boiled down example, 
+   in your app you should remove the event listener on service teardown.
+   
+   ```js
+   // app/services/network-connection.js
+   
+   import Service, { inject as service } from '@ember/service';
+
+   export default Service.extend({
+     router: service(),
+     
+     init() {
+       this._super();
+       
+       window.addEventListener('offline', () => {
```

Comment:
Should we not take down the event, even if Services are singletons?

Reply:
It's an example, I think this code is sufficient for it's purpose (explaining what the router server is and how it can be used). In my opinion adding event teardown code would make the example needlessly long.

Reply:
@sandstrom would it help to link to ember observer to show different use cases https://emberobserver.com/code-search?codeQuery=router%3A%20service() ?

Reply:
Sure, feel free to [amend](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) this PR/brach as you see fit.